### ISSUE TITLE
Denying or restricting contact permission causes the app to keep crashing.

### ISSUE THREAD

**Issue by organik1** [TeamAmaze/AmazeFileManager#1248]
Hello dev

I noticed that whenever I restrict contact permission for the app on my 6.0 marshmallow device, the app crashes repeatedly.  Maybe the app wouldn't run without the permission but a toast informing the user about the essence of this permission would be much preferable to a crash.

**Comment by organik1** [comment]
Update!!

This happens after signing in with Google+

**Comment by TranceLove** [comment]
This is weird... we should never asked for contact permission at all.

![screenshot_1526980512](https://user-images.githubusercontent.com/251819/40353342-dd1988c8-5de3-11e8-9bc4-0145dd671dbf.png)
(Pixel 2 emulator running Android 6.0/API 23)

**Comment by VishalNehra** [comment]
On play store version, we do ask for contact permissions once you choose 'sign in with g+' from settings. It is necessary to get your name and profile picture for G+

**Comment by organik1** [comment]
TranceLove, I could upload some screenshots from my end, if you want

**Comment by TranceLove** [comment]
Actually logcat message will be more useful. Anyway I'll try to find out when I got back.

(Suddenly I realised I'm so far away from Earth...)

**Comment by organik1** [comment]
Lols! Be back soon. I'll make the logcat available soon.

**Comment by EmmanuelMess** [comment]
This should be fixed in version 1.3. Haven't removed G+ connection and the permission?

**Comment by TranceLove** [comment]
Confirmed with Play Store's current release 3.2.1, on Pixel 2 emulator running Android 6.0/API 23. When my G+ has a photo, after I opt-in to sign in with Google, whenever I launch Amaze it'll crash.

Logcat said
```
    java.lang.SecurityException: Missing android.permission.GET_ACCOUNTS
        at android.os.Parcel.readException(Parcel.java:1599)
        at android.os.Parcel.readException(Parcel.java:1552)
        at com.google.android.gms.plus.internal.zzd$zza$zza.getAccountName(Unknown Source)
        at com.google.android.gms.plus.internal.zze.getAccountName(Unknown Source)
        at com.google.android.gms.internal.zzazb.getAccountName(Unknown Source)
        at com.amaze.filemanager.activities.MainActivity.onConnected(MainActivity.java:1562)
        at com.google.android.gms.common.internal.zzm.zzq(Unknown Source)
        at com.google.android.gms.internal.zzaat.zzo(Unknown Source)
        at com.google.android.gms.internal.zzaar.zzwi(Unknown Source)
        at com.google.android.gms.internal.zzaar.onConnected(Unknown Source)
        at com.google.android.gms.internal.zzaav.onConnected(Unknown Source)
        at com.google.android.gms.internal.zzaag.onConnected(Unknown Source)
        at com.google.android.gms.common.internal.zzl$1.onConnected(Unknown Source)
        at com.google.android.gms.common.internal.zzf$zzj.zzxG(Unknown Source)
        at com.google.android.gms.common.internal.zzf$zza.zzb(Unknown Source)
        at com.google.android.gms.common.internal.zzf$zza.zzu(Unknown Source)
        at com.google.android.gms.common.internal.zzf$zze.zzxH(Unknown Source)
        at com.google.android.gms.common.internal.zzf$zzd.handleMessage(Unknown Source)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by organik1** [comment]
I see you've helped yourself with the log. When will a new version be available..??

**Comment by VishalNehra** [comment]
You should allow contacts permission for crash to not occur.

**Comment by organik1** [comment]
Yes yes; that I have done.

**Comment by EmmanuelMess** [comment]
> This should be fixed in version 1.3. Haven't removed G+ connection and the permission?

Waiting confirmation, this should be closed.

**Comment by TranceLove** [comment]
> Waiting confirmation

@EmmanuelMess you mean this?

> However, as of d479548 (date marked November 25 2017) Google Plus integration was removed... so this should not happen on master, because the integration was removed.
